### PR TITLE
Update sparspak

### PR DESCRIPTION
**Pull request by j-fu**

### CODE REVIEW

**Comment by PetrKryslUCSD** [comment]
This looks good. Agreed on the precompilation.

**Comment by ChrisRackauckas** [comment]
Precompiling for Float64, which is by far the most common case, is probably useful though.

File: src/factorization.jl [hunk 1]
```diff
@@ -503,30 +503,28 @@ end
...
 function init_cacheval(::SparspakFactorization, A, b, u, Pl, Pr, maxiters::Int, abstol,
                        reltol,
                        verbose::Bool, assumptions::OperatorAssumptions)
     A = convert(AbstractMatrix, A)
-    p = Sparspak.Problem.Problem(size(A)...)
-    Sparspak.Problem.insparse!(p, A)
-    Sparspak.Problem.infullrhs!(p, b)
-    s = Sparspak.SparseSolver.SparseSolver(p)
-    return s
+    return sparspaklu(A)
```
Comment (ChrisRackauckas) [inline src/factorization.jl:514]: this performs an extra lu-factorization though? Can we avoid this and just build a blank/empty LU object to fill later?

Comment (j-fu) [reply src/factorization.jl:514]: I see. Assuming I have all info on A this is possible but not in the API. I think I could  add a kwarg to sparspaklu.

Comment (ChrisRackauckas) [reply src/factorization.jl:514]: I think a kwarg would be a nice solution.

**Review by ChrisRackauckas — APPROVED** [review]

**Comment by ChrisRackauckas** [comment]
I'll run the formatter, thanks!

**Pull request merged**